### PR TITLE
[PB] Map Component

### PR DESCRIPTION
**Pull request by danmooney2**
## Description

Map content type in Venia UI / PWA Studio

## Related Issue

https://jira.corp.magento.com/browse/PB-19

## Performance Results
https://wiki.corp.magento.com/display/COREENG/Map

## Verification Steps

1. Check that each property for each content type appearance renders correctly on PWA Venia storefront in desktop and mobile.
2. Compare Venia PWA rendering of content type with Luma to ensure the experience is the same or better for each property of each appearance in desktop and mobile.
3. Check each supported browser and on mobile that is supported according to PWA documentation.
4. Acceptable performance has been validated

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/pwa-studio?pullRequest=1790) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/pwa-studio?pullRequest=1790) before we can accept your contribution.<br/>

File: packages/venia-ui/lib/components/RichContent/PageBuilder/ContentTypes/Map/__mocks__/loadGoogleMapsApi.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    GoogleMapsLatLngBoundsConstructor: jest.fn(() => ({
+        extend: mocks.googleMapsLatLngBoundsInstanceExtend
+    })),
+    GoogleMapsInfoWindowConstructor: jest.fn(() => ({
+        open: mocks.googleMapsInfoWindowInstanceOpen,
+        close: mocks.googleMapsInfoWindowInstanceClose
+    }))
+};
+
+mocks.googleMapsInstance = {
```
Comment (davemacaulay) [inline packages/venia-ui/lib/components/RichContent/PageBuilder/ContentTypes/Map/__mocks__/loadGoogleMapsApi.js]: Isn't this already part of the mock from the const above?

File: packages/venia-ui/lib/components/RichContent/PageBuilder/ContentTypes/Map/map.js [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    return (
+        <div
+            ref={mapElement}
+            style={dynamicStyles}
+            className={[defaultClasses.root, ...cssClasses].join(' ')}
+        />
+    );
+};
+
+Map.propTypes = {
```
Comment (davemacaulay) [inline packages/venia-ui/lib/components/RichContent/PageBuilder/ContentTypes/Map/map.js]: You need to JSDoc comment this for inclusion in the documentation.

File: packages/venia-ui/lib/components/RichContent/PageBuilder/ContentTypes/Map/__mocks__/loadGoogleMapsApi.js (hunk 1 above)
Comment (danmooney2) [reply packages/venia-ui/lib/components/RichContent/PageBuilder/ContentTypes/Map/__mocks__/loadGoogleMapsApi.js]: haha 💯

**Pull request merged**